<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PostMachineView</class>
 <widget class="QMainWindow" name="PostMachineView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>798</width>
    <height>719</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Post Machine</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../data/recources.qrc">
    <normaloff>:/icons/postmachine.png</normaloff>:/icons/postmachine.png</iconset>
  </property>
  <property name="locale">
   <locale language="Ukrainian" country="Ukraine"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="locale">
    <locale language="English" country="UnitedStates"/>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QToolButton" name="run_button">
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>30</height>
           </size>
          </property>
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="stop_button">
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>30</height>
           </size>
          </property>
          <property name="text">
           <string>Stop</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="step_button">
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>30</height>
           </size>
          </property>
          <property name="text">
           <string>Next</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="reset_button">
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>30</height>
           </size>
          </property>
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Problem</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="problem_edit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>80</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="tape_layout">
        <item>
         <widget class="QPushButton" name="scroll_left_button">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>30</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>&lt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="TapeWidget" name="tape_widget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>600</width>
            <height>80</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="scroll_right_button">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>30</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QToolButton" name="insert_button">
            <property name="minimumSize">
             <size>
              <width>30</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>+</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="remove_button">
            <property name="minimumSize">
             <size>
              <width>30</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableWidget" name="commands_widget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>10</verstretch>
           </sizepolicy>
          </property>
          <property name="autoScroll">
           <bool>true</bool>
          </property>
          <property name="showGrid">
           <bool>true</bool>
          </property>
          <property name="gridStyle">
           <enum>Qt::NoPen</enum>
          </property>
          <property name="sortingEnabled">
           <bool>false</bool>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="cornerButtonEnabled">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderVisible">
           <bool>true</bool>
          </attribute>
          <attribute name="horizontalHeaderCascadingSectionResizes">
           <bool>false</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderCascadingSectionResizes">
           <bool>false</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Command</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Jump</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Comment</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>798</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="new_file_action"/>
    <addaction name="load_action"/>
    <addaction name="save_action"/>
    <addaction name="separator"/>
    <addaction name="exit_action"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Tape</string>
    </property>
    <addaction name="load_tape_action"/>
    <addaction name="save_tape_action"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="instruction_action"/>
    <addaction name="about_action"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>Speed</string>
    </property>
    <addaction name="fast_speed_action"/>
    <addaction name="normal_speed_action"/>
    <addaction name="slow_speed_action"/>
    <addaction name="separator"/>
    <addaction name="custom_speed_action"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_4"/>
   <addaction name="menu_3"/>
  </widget>
  <action name="instruction_action">
   <property name="text">
    <string>Guide</string>
   </property>
   <property name="shortcut">
    <string>G</string>
   </property>
  </action>
  <action name="about_action">
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>A</string>
   </property>
  </action>
  <action name="load_tape_action">
   <property name="text">
    <string>Load</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+O</string>
   </property>
  </action>
  <action name="save_tape_action">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+S</string>
   </property>
  </action>
  <action name="load_action">
   <property name="text">
    <string>Load</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="save_action">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="exit_action">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="fast_speed_action">
   <property name="text">
    <string>Fast</string>
   </property>
   <property name="shortcut">
    <string>F</string>
   </property>
  </action>
  <action name="normal_speed_action">
   <property name="text">
    <string>Normal</string>
   </property>
   <property name="shortcut">
    <string>N</string>
   </property>
  </action>
  <action name="slow_speed_action">
   <property name="text">
    <string>Slow</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="custom_speed_action">
   <property name="text">
    <string>Custom</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="new_file_action">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TapeWidget</class>
   <extends>QWidget</extends>
   <header>tapewidget.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../data/recources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>exit_action</sender>
   <signal>changed()</signal>
   <receiver>PostMachineView</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
